src/20230926_AaveV3_Eth_GHOFunding/AaveV3_Ethereum_GHOFunding_20230926.sol

```diff
@@ -0,0 +1,116 @@
+ */
+contract AaveV3_Ethereum_GHOFunding_20230926 is IProposalGenericExecutor {
+
+  struct Asset {
+    address underlying;
+    address aToken;
+    address oracle;
+    uint256 amount;
+  }
+
+  address public constant COLLECTOR = address(AaveV3Ethereum.COLLECTOR);
+  address public constant MILKMAN = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public constant PRICE_CHECKER = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+  address public constant GHO_ORACLE = 0x3f12643D3f6f874d39C2a4c9f2Cd6f2DbAC877FC; 
+
+  function execute() external {
+    // DAI v2
+    Asset memory DAIv2 = Asset({
+      underlying: AaveV2EthereumAssets.DAI_UNDERLYING,
+      aToken: AaveV2EthereumAssets.DAI_A_TOKEN,
+      oracle: 0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9,
+      amount: IERC20(AaveV2EthereumAssets.DAI_A_TOKEN).balanceOf(COLLECTOR)
+    });
+
+    // DAI v3
+    Asset memory DAIv3 = Asset({
+      underlying: AaveV3EthereumAssets.DAI_UNDERLYING,
+      aToken: AaveV3EthereumAssets.DAI_A_TOKEN,
+      oracle: 0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9,
+      amount: 370_000 * 1e18
+    });
+
+    // USDT v2
+    Asset memory USDTv2 = Asset({
+      underlying: AaveV2EthereumAssets.USDT_UNDERLYING,
+      aToken: AaveV2EthereumAssets.USDT_A_TOKEN,
+      oracle: 0x3E7d1eAB13ad0104d2750B8863b489D65364e32D,
+      amount: 370_000 * 1e6
+    });
+
+    // BUSD
+    Asset memory BUSD = Asset({
+      underlying: AaveV2EthereumAssets.BUSD_UNDERLYING,
+      aToken: address(0), // not used
+      oracle: 0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A,
+      amount: IERC20(AaveV2EthereumAssets.BUSD_UNDERLYING).balanceOf(COLLECTOR)
+    });
+    
+    // TUSD v2
+    Asset memory TUSD = Asset({
+      underlying: AaveV2EthereumAssets.TUSD_UNDERLYING,
+      aToken: address(0), // not used
+      oracle: 0xec746eCF986E2927Abd291a2A1716c940100f8Ba,
+      amount: IERC20(AaveV2EthereumAssets.TUSD_UNDERLYING).balanceOf(COLLECTOR)
+    });
+
+
+    ////// DAI v2 & v3 swap //////
+    AaveV3Ethereum.COLLECTOR.transfer(DAIv2.aToken, address(this), DAIv2.amount);
+    AaveV3Ethereum.COLLECTOR.transfer(DAIv3.aToken, address(this), DAIv3.amount);
+    uint256 swapperBalance = 
+      AaveV2Ethereum.POOL.withdraw(DAIv2.underlying, DAIv2.amount, AaveMisc.AAVE_SWAPPER_ETHEREUM);
+    swapperBalance += 
+      AaveV3Ethereum.POOL.withdraw(DAIv3.underlying, DAIv3.amount, AaveMisc.AAVE_SWAPPER_ETHEREUM);
+    swapAsset(DAIv2, swapperBalance);
+
+    ////// BUSD swap //////
+    AaveV3Ethereum.COLLECTOR.transfer(BUSD.underlying, AaveMisc.AAVE_SWAPPER_ETHEREUM, BUSD.amount);
+    swapAsset(BUSD, BUSD.amount);
+
+    ////// TUSD swap //////
+    AaveV3Ethereum.COLLECTOR.transfer(TUSD.underlying, AaveMisc.AAVE_SWAPPER_ETHEREUM, TUSD.amount);
+    swapAsset(TUSD, TUSD.amount);
+
+    ////// USDT v2 swap //////
+    AaveV3Ethereum.COLLECTOR.transfer(USDTv2.aToken, address(this), USDTv2.amount);
+    swapperBalance = 
```

Comment:
is the gas saving really worth it?

why not just use descriptive variable names - e.g. `daiBalance` & `usdtBalance`